kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/ProcessResource.java

```diff
@@ -328,9 +329,20 @@ public Response startProcessWithCorrelationKeyFromNodeIds(@javax.ws.rs.core.Cont
         } catch (SecurityException e) {
             return forbidden(errorMessage(e, e.getMessage()), v, conversationIdHeader);
         } catch (Exception e) {
-            logger.error("Unexpected error during processing {}", e.getMessage(), e);
-            return internalServerError(
-                                       MessageFormat.format(CREATE_RESPONSE_ERROR, e.getMessage()), v);
+            Throwable t = e.getCause();
+            SQLIntegrityConstraintViolationException constraintViolationException = null;
+
+            while ((t != null) && !(t instanceof SQLIntegrityConstraintViolationException)) {
+                t = t.getCause();
+                constraintViolationException = (SQLIntegrityConstraintViolationException) t;
+            }
+            if (constraintViolationException != null && (constraintViolationException.getSQLState() == "23000" || e.getMessage().contains("already exists"))) {
```

Comment:
I think it would be better to keep database specific code encapsulated in jBPM.
Also, is the SQL state `23000` and `already exists` message same in all supported databases?

Reply:
Hi. there is no place were you could place that code without affecting the engine behaviour. So setting this a jbpm engine level is risky. starting a process is something that happens at more levels that just the main process.
Regarding the state and message I agree.

Reply:
@sutaakar For 'already exists' keyword is from [JpaProcessPersistenceContext](https://github.com/kiegroup/jbpm/blob/main/jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/JpaProcessPersistenceContext.java#L126) code so it should work for all DB's. Regarding '23000' SQLState, from supported DB list,  [MSSQL](https://docs.microsoft.com/en-us/sql/odbc/reference/appendixes/appendix-a-odbc-error-codes?view=sql-server-ver15),[Mysql](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-error-sqlstates.html),[Oracle](https://docs.oracle.com/cd/E15817_01/appdev.111/b31228/appd.htm),[Postgres](http://www.ing.iac.es/~docs/external/postgres/html/errcodes-appendix.html) and [MariaDB](https://mariadb.com/kb/en/mariadb-error-codes)  uses 2300 SQLState constraint violation.  IBM DB2 and Sybase DB uses different SQLState, for these DB's 'already exists' keyword should work.

Reply:
I am not a big fan of relying on specific implementation details from inner layer (specific message in exception).
Anyway, please at least add a comment in the code, explaining that `already exists` reference to the exception message from JpaProcessPersistenceContext.

Also, can you please add an integration test checking the desired behaviour?

Reply:
Integration test added